core/domain/rights_manager.py

```diff
@@ -273,23 +324,41 @@ def _get_activity_rights_where_user_is_owner(activity_type, user_id):
         list(ActivityRights). List of domain objects where the user has some
         role.
     """
+    # Ruling out the possibility of any other activity type.
```

Comment:
Per previous discussion, either:

- do an if/exception, and the assert after that
- or: validate that the activity_type is one of the given two specific things using mypy, if that is possible.

Reply:
I think we already discussed this in a comment https://github.com/oppia/oppia/pull/15693#discussion_r919937597.
If we use `if clause` then that would never be executed and I have to write the test for this private method individually.

Reply:
Ah, thanks -- yeah, agreed, the assertion is better than what existed before, and I get your point about not wanting to test a private method for this. I'm fine with that.

Just to check though. Is there a way to more strictly check that the activity_type matches one of the given string literals using mypy?

Reply:
> Is there a way to more strictly check that the activity_type matches one of the given string literals using mypy?

No, Reason:
In many cases, we are providing `constants.ACTIVITY_TYPE_EXPLORATION` or `constants.ACTIVITY_TYPE_COLLECTION`, and all these `constants`'s constants have values only at runtime. So, MyPy is not able to fetch that these are constants having a particular Literal strings.